babel.config.js

```diff
@@ -25,52 +25,218 @@
+    if (experimentDefine && !replacements.some(flagExists)) {
+      replacements.push(createReplacement(experimentDefine, false));
+    }
+  });
+
+  // default each backup experiment constant to the customized value
+  // eslint-disable-next-line local/no-for-of-statement, local/no-array-destructuring
+  for (const [experimentDefine, value] of Object.entries(
+    experimentsConstantBackup
+  )) {
+    // eslint-disable-next-line require-jsdoc
+    function flagExists(element) {
+      return element['identifierName'] === experimentDefine;
+    }
+
+    // only add default replacement if it already doesn't exist in array
+    if (experimentDefine && !replacements.some(flagExists)) {
+      replacements.push(createReplacement(experimentDefine, !!value));
+    }
+  }
+
+  return ['minify-replace', {replacements}];
+}
+
+const eliminateIntermediateBundles = () => [
+  localPlugin('transform-prune-namespace'),
 ];
 
+const getJsonConfigurationPlugin = () =>
+  localPlugin('transform-json-configuration');
+
+/**
+ * Resolves babel plugin set to apply before compiling on singlepass.
+ * @param {!Object<string, boolean>} buildFlags
+ * @return {!Array<string|!Array<string|!Object>>}
+ */
+function plugins(buildFlags) {
+  const {isEsmBuild, isForTesting, isSinglePass, isChecktypes} = buildFlags;
+  // eslint-disable-next-line local/no-spread
+  const applied = [...defaultPlugins(isEsmBuild || false)];
+  // TODO(erwinm): This is temporary until we remove the assert/log removals
+  // from the java transformation to the babel transformation.
+  // There is currently a weird interaction where when we do the transform
+  // in babel and leave a bare "string", Closure Compiler does not remove
+  // the dead string expression statements. We cannot just outright remove
+  // the argument of the assert/log calls since we would need to inspect
+  // if the arguments have any method calls (which might have side effects).
+  if (isSinglePass) {
+    applied.push(localPlugin('transform-amp-asserts'));
+  }
+  if (isEsmBuild) {
+    applied.push([
+      'filter-imports',
+      {
+        imports: {
+          ...esmRemovedImports,
+          ...validTransformedRemovableImports,
+        },
+      },
+    ]);
+  }
+  if (isChecktypes) {
+    applied.push(localPlugin('transform-simple-object-destructure'));
+  } else {
+    // This triggers some conformance errors such as `use tryParseJson` during
+    // type check phase so we omit it from the default plugins.
+    applied.push(getJsonConfigurationPlugin());
+  }
+  if (!(isForTesting || isChecktypes)) {
+    applied.push(
+      localPlugin('amp-mode-transformer'),
+      localPlugin('is_dev-constant-transformer')
+    );
+  }
+  return applied;
+}
+
 // eslint-disable-next-line local/no-module-exports
 module.exports = function(api) {
+  const isClosureCompiler =
+    argv._.includes('dist') || argv._.includes('check-types');
+  const {esm} = argv;
+  const noModuleTarget = {
+    'browsers': isTravisBuild()
```

Comment:
We only want this if in single-pass.

Reply:
Are you saying that's how it works now and something has changed, or that's not how it works now but it should be fixed? [[original code](https://github.com/ampproject/amphtml/blob/master/babel.config.js#L35)]